L1Trigger/Phase2L1Taus/src/L1HPSPFTauBuilder.cc

```diff
@@ -0,0 +1,446 @@
+#include "L1Trigger/Phase2L1Taus/interface/L1HPSPFTauBuilder.h"
+#include "FWCore/Utilities/interface/Exception.h"  // cms::Exception
+#include "DataFormats/Math/interface/deltaR.h"     // reco::deltaR
+#include <TString.h>                               // TString, Form()
+#include <TMath.h>                                 // TMath::Pi()
+#include <string>                                  // std::string
+#include <algorithm>                               // std::max(), std::sort()
+#include <cmath>                                   // std::fabs
+
+L1HPSPFTauBuilder::L1HPSPFTauBuilder(const edm::ParameterSet& cfg)
+    : signalConeSizeFormula_(nullptr),
+      minSignalConeSize_(cfg.getParameter<double>("minSignalConeSize")),
+      maxSignalConeSize_(cfg.getParameter<double>("maxSignalConeSize")),
+      useStrips_(cfg.getParameter<bool>("useStrips")),
+      stripSizeEta_(cfg.getParameter<double>("stripSizeEta")),
+      stripSizePhi_(cfg.getParameter<double>("stripSizePhi")),
+      isolationConeSize_(cfg.getParameter<double>("isolationConeSize")),
+      debug_(cfg.getUntrackedParameter<bool>("debug", false)) {
+  std::string signalConeSizeFormulaOriginal = cfg.getParameter<std::string>("signalConeSize");
+  TString signalConeSizeFormulaModified = signalConeSizeFormulaOriginal.data();
+  signalConeSizeFormulaModified.ReplaceAll("pt", "x");
+  signalConeSizeFormula_ = std::make_unique<reco::FormulaEvaluator>(signalConeSizeFormulaModified.Data());
```

Comment:
@cecilecaillol @rekovic can you double check with RECO the cost (memory, runtime compilation, etc.) and performance of `reco::FormulaEvaluator` ?

Reply:
also - why an `std::unique_ptr` instead of having the `reco::FormulaEvaluator` directly as a data member ?